### PR TITLE
Plans: update domain to plan nudge to use Banner component

### PR DESCRIPTION
**Pull request by gwwar**
This PR updates the domain to plan nudge to use a [Banner](https://wpcalypso.wordpress.com/devdocs/design/banner) component. h/t @Copons and @folletto 

See previous related PRs: #8635 and #9254

Other copy suggestions are welcome.

Desktop:
screen shot 2017-01-17 at 10 52 07 am

Mobile:
screen shot 2017-01-17 at 10 52 19 am

### Testing Instructions:
- Navigate to http://calypso.localhost:3000/domains
- Select a wpcom site that has a paid domain, but a free plan
- We should see a nudge render below domains
- An impression analytics event should fire:
screen shot 2017-01-16 at 3 59 00 pm
- Clicking on the nudge button takes us to checkout with a personal plan in cart
- The following analytics event should fire:
screen shot 2017-01-16 at 3 59 23 pm

cc @rralian @sararosso @apeatling

### CODE REVIEW

File: client/blocks/domain-to-plan-nudge/index.jsx [hunk 1]
```diff
@@ -67,149 +58,42 @@ class DomainToPlanNudge extends Component {
...
+				}
+				event="domain_to_personal_nudge" //actually cta_name
+				dismissPreferenceName="domain-to-plan-nudge"
+				feature={ FEATURE_NO_ADS }
+				href={ `/checkout/${ siteId }/personal` }
+				list={
+					[
+						translate( 'No advertising' ),
+						translate( 'Live chat support' ),
+						translate( 'Bundled with your domain for best value!' )
```
Comment (a8ci18n) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Alternate string suggestion: 
* `translate( 'Bundled with your domain for the best value!' )`&mdash; translations: **17**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-to-plan-nudge)

File: client/blocks/domain-to-plan-nudge/index.jsx [hunk 2]
```diff
@@ -67,149 +58,42 @@ class DomainToPlanNudge extends Component {
...
+						comment: '%s will be replaced by a formatted price, i.e $9.99'
+					} )
+				}
+				event="domain_to_personal_nudge" //actually cta_name
+				dismissPreferenceName="domain-to-plan-nudge"
+				feature={ FEATURE_NO_ADS }
+				href={ `/checkout/${ siteId }/personal` }
+				list={
+					[
+						translate( 'No advertising' ),
```
Comment (a8ci18n) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Alternate string suggestion: 
* `translate( 'No Advertising' )`&mdash; translations: **25**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-to-plan-nudge)

File: client/blocks/domain-to-plan-nudge/index.jsx [hunk 3]
```diff
@@ -67,149 +58,42 @@ class DomainToPlanNudge extends Component {
...
-									comment: '%s will be replaced by a formatted price, i.e $9.9'
-								} )
-							}
-						</Button>
-					</div>
-				</div>
-			</DismissibleCard>
+			<Banner
+				callToAction={
+					translate( 'Upgrade for %s', {
```
Comment (a8ci18n) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Alternate string suggestion: 
* `translate( 'Upgrades for %s' )`&mdash; translations: **25**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-to-plan-nudge)

File: client/blocks/domain-to-plan-nudge/index.jsx (hunk 1 above)
Comment (gwwar) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Bot caught my typo 😄

File: client/blocks/domain-to-plan-nudge/index.jsx (hunk 2 above)
Comment (a8ci18n) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Alternate string suggestion: 
* `translate( 'No Advertising' )`&mdash; translations: **25**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-to-plan-nudge)

File: client/blocks/domain-to-plan-nudge/index.jsx (hunk 3 above)
Comment (a8ci18n) [inline client/blocks/domain-to-plan-nudge/index.jsx]: Alternate string suggestion: 
* `translate( 'Upgrades for %s' )`&mdash; translations: **25**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-to-plan-nudge)

**Comment by folletto** [comment]
Design wise looks good, 👍  

I'm just wondering if this isn't too jargon-y:

> Bundled with your domain for the best value!

Maybe we can rephrase it as:

> Includes your domain of choice

Or something similar?

**Comment by Copons** [comment]
Code wise LGTM!

**Comment by apeatling** [comment]
LGTM!

**Comment by sararosso** [comment]
Any reason not to use the same copy on the /plans page? 

I would substitute for the copy: 
- Remove WordPress.com ads
- Email and live chat support

The last point is a bit unclear for the user - some alternatives: 
- Use with your current custom domain
- Works together with your custom domain 
etc.

We could also include "- Extra 3GB space" but will probably be too long.

And I would make "save" capitalized as well - "Upgrade to a Personal Plan and Save!"

**Comment by sararosso** [comment]
> Includes your domain of choice

I wouldn't use this copy - this is for users who have already chosen a domain and they won't be given a domain credit to choose a new domain.

**Comment by gwwar** [comment]
Thanks @sararosso I've updated the summary with new screenshots of the suggested copy.

**Comment by gwwar** [comment]
I'll land this tomorrow if folks are happy with the copy here.

**Comment by sararosso** [comment]
> I'd probably then replace the third line about the domain (the user already has it, so there's no point in mentioning it) with the "Extra 3GB space" line.

Hm, yeah, I'm torn here - I think the "works with your current domain" concept is important so it reinforces to people that they don't have to change their site address / domain etc. to get it, but that also may be how we see it from the inside - the Personal Plan normally includes a domain credit; this version wouldn't.

**Comment by folletto** [comment]
> Hm, yeah, I'm torn here - I think the "works with your current domain" concept is important so it reinforces to people that they don't have to change their site address / domain etc. to get it, but that also may be how we see it from the inside - the Personal Plan normally includes a domain credit; this version wouldn't.

I had the same doubt. The reasoning I followed was:

1. If I'm a user, on that page, I already know what I have, and then I see an "Upgrade" message. Since I don't know what the plan would give me, there's no need to "negate" the domain offering there.
2. The next page does't mention any detail anymore, so we're not adding something.
3. If the user digs for more information, they will still get the "Custom Domain Name" message.  
4. Having more space is still an extra.

So yes, in practice I'm implying the user isn't already aware what the Personal Plan offers.

If that's a concern, then I'm ok in keeping it. :)

**Comment by gwwar** [comment]
Maybe lets stick with this version for now. Thanks for the reviews!

**Pull request merged**